src/lib/parsers/sol-json.c

```diff
@@ -789,6 +790,492 @@ sol_json_serialize_boolean(struct sol_buffer *buffer, bool val)
+        if (opts->structure.separator.len) {
+            r = sol_buffer_append_slice(buf, opts->structure.separator);
+            if (r < 0)
+                return r;
+        }
+    }
+
+    r = memdesc_serialize_structure_member_key(member, buf, opts, prefix);
+    if (r < 0)
+        return r;
+
+    return sol_memdesc_serialize(&member->base, memory, buf, opts, prefix);
+}
+
+SOL_API int
+sol_json_serialize_memdesc(struct sol_buffer *buffer, const struct sol_memdesc *desc, const void *memory, bool detailed_structures)
+{
+    const struct sol_memdesc_serialize_options json_serialize_options = {
+        SOL_SET_API_VERSION(.api_version = SOL_MEMDESC_SERIALIZE_OPTIONS_API_VERSION, )
+        .serialize_int64 = memdesc_serialize_int64,
+        .serialize_uint64 = memdesc_serialize_uint64,
+        .serialize_double = memdesc_serialize_double,
+        .serialize_boolean = memdesc_serialize_boolean,
+        .serialize_pointer = memdesc_serialize_pointer,
+        .serialize_string = memdesc_serialize_string,
+        .serialize_structure_member = memdesc_serialize_structure_member,
+        .structure = {
+            .container = {
+                .start = SOL_STR_SLICE_LITERAL("{"),
+                .end = SOL_STR_SLICE_LITERAL("}"),
+            },
+            .key = {
+                .end = SOL_STR_SLICE_LITERAL(":"),
+            },
+            .separator = SOL_STR_SLICE_LITERAL(","),
+            .show_key = true,
+            .detailed = detailed_structures,
+        },
+        .array = {
+            .container = {
+                .start = SOL_STR_SLICE_LITERAL("["),
+                .end = SOL_STR_SLICE_LITERAL("]"),
+            },
+            .separator = SOL_STR_SLICE_LITERAL(","),
+            .show_index = false,
+        },
+    };
+
+    SOL_NULL_CHECK(buffer, -EINVAL);
+    SOL_NULL_CHECK(desc, -EINVAL);
+    SOL_NULL_CHECK(memory, -EINVAL);
+
+    return sol_memdesc_serialize(desc, memory, buffer, &json_serialize_options, NULL);
+}
+
+static inline const struct sol_memdesc_structure_member *
+find_memdesc_for_key(const struct sol_memdesc *desc, const struct sol_json_token *token)
+{
+    const struct sol_memdesc_structure_member *found;
+    struct sol_buffer buf;
+    struct sol_str_slice key;
+
+    if (sol_json_token_get_unescaped_string(token, &buf) < 0)
+        return NULL;
+
+    key = sol_buffer_get_slice(&buf);
+    found = sol_memdesc_find_structure_member(desc, key);
+    sol_buffer_fini(&buf);
+    return found;
+}
+
+SOL_API int
+sol_json_load_memdesc(const struct sol_json_token *token, const struct sol_memdesc *desc, void *memory)
+{
+    int r;
+
+    SOL_NULL_CHECK(token, -EINVAL);
+    SOL_NULL_CHECK(desc, -EINVAL);
+    SOL_NULL_CHECK(memory, -EINVAL);
+
+    switch (desc->type) {
+    case SOL_MEMDESC_TYPE_UNKNOWN:
+        return -EINVAL;
+    case SOL_MEMDESC_TYPE_UINT8: {
+        uint8_t *m = memory;
+        uint64_t v;
+
+        r = sol_json_token_get_uint64(token, &v);
```

Comment:
Maybe it's a silly question, but if we have `sol_json_token_get_uint32`, why use `sol_json_token_get_uint64` when getting things < 32?